AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -210,8 +229,18 @@ public final class APIClient: APIClientProtocol {
     private func handle<R: Request>(
         _ result: URLSessionDownloadSuccess,
         _ request: R
-    ) -> HTTPResponse<R.ResponseType> where R.ResponseType == DownloadResponse {
+    ) throws -> HTTPResponse<R.ResponseType> where R.ResponseType == DownloadResponse {
         log(result: result, request: request)
+        
+        
+        try DispatchQueue.global(qos: .default).sync {
+            assert(!Thread.isMainThread, "Should not be running on main thread!!")
```

Comment:
also this assert will be triggered sometimes because global queue might use the main thread sometimes as an optimization as discussed offline